### DISCUSSION TITLE
Performance issue

### DISCUSSION THREAD

**Discussion by sathyarajshetigar** [TomSchimansky/CustomTkinter#1119] [General]
Hi,
I built a trading dashboard as shown below.
It sometimes performs very poorly and becomes unresponsive.
When I profile it using pycharm it runs alright. When I make the executable using pyInstaller it runs slow sometimes and hangs
Have a look at the image and tell if I am using way too many widgets!
and what else I can consider to improve the performance!
Sharing the pystat file as well if dev team can analyse it.
https://drive.google.com/file/d/1V20iRk1ulw8KHbPoplV8AY304nFP5F9X/view?usp=sharing

**Comment by sumeshir26** [comment]
While using pysinstaller try without the --onefile option

**Reply by sathyarajshetigar** [reply]
@sumeshir26  I tried that as well.  I feel after upgrading to 5.0.5 app is become slow

**Comment by TomSchimansky** [comment]
I think I have to see your code an run it for myself to see where the problem is. What kind of widget is at the bottom in your screenshot, a CTkTextbox or tkinter.Text?

**Reply by sathyarajshetigar** [reply]
Hi,
Sharing the code block where it updates the tkinter.Text
I get the trade positions data through websocket and I construct a DF and send to the following method
` def onPositionUpdate(self, df):
        try:
            if df is not None:
                if not df.empty:
                    positionsDF = df
                    data = positionsDF.values.tolist()
                    self.positionsTable.clear_rows()
                    self.positionsTable.field_names = positionsDF.columns.tolist()
                    self.positionsTable.min_width = int((self.winfo_width() / 7) / 13)  # 6 columns
                    self.positionsTable.set_style(style=16)
                    # Add the data to the table
                    for row in data:
                        self.positionsTable.add_row(row)
                    self.positionsLabel.configure(state="normal")
                    self.positionsLabel.delete("1.0", tkinter.END)  # Delete the current text
                    self.positionsLabel.insert("1.0", f"{self.positionsTable}")  # Insert the new text at the
                    self.positionsLabel.configure(state="disabled")
            else:
                # print('onPositionUpdateInThread df is empty')
                if self.positionsLabel is not None:
                    self.positionsLabel.configure(state="normal")
                    self.positionsLabel.delete("1.0", tkinter.END)  # Delete the current text
                    self.positionsLabel.configure(state="disabled")
        except Exception as e:
            Globals.log(f'onPositionUpdateInThread Error {e}', sendToTelegram=True)`
            
When DF is empty if clear the text field.

**Reply by sathyarajshetigar** [reply]
More readable 
![image](https://user-images.githubusercontent.com/42057975/215243436-40c52d54-0d4e-4391-9082-ed8ed3617a15.png)

**Comment by sathyarajshetigar** [comment]
I removed the Text and shifted to treeview now the performance is alright
